src/_pytest/config/__init__.py

```diff
@@ -1215,6 +1215,11 @@ def create_terminal_writer(config: Config, *args, **kwargs) -> TerminalWriter:
         tw.hasmarkup = True
     if config.option.color == "no":
         tw.hasmarkup = False
+
+    if config.option.code_highlight == "no":
+        tw.code_highlight = False
+    else:
```

Comment:
Let's change this to `config.option.code_highlight == "yes"`, to be consistent with the `hasmarkup` code above. I'd also switch the order such that `"yes"` is checked first, for the same reason.